evap/staff/importers.py

```diff
@@ -497,26 +497,51 @@ def write_enrollments_to_db(self, semester, vote_start_datetime, vote_end_date):
             # add all participants at once to create only a single log message
             for evaluation, participants in participants_per_evaluation.items():
                 evaluation.participants.add(*participants)
+            self.create_success_messages(students_created, responsibles_created)
 
-        msg = format_html(
-            _("Successfully created {} courses/evaluations, {} participants and {} contributors:"),
-            len(self.evaluations),
-            len(students_created),
-            len(responsibles_created),
+    def create_success_messages(self, students_created, responsibles_created):
+        student_count = len(students_created)
+        student_word = pgettext("student count", "no") if student_count == 0 else str(student_count)
+        responsible_count = len(responsibles_created)
+        responsible_word = pgettext("responsible count", "no") if responsible_count == 0 else str(responsible_count)
+
+        student_phrase = ngettext("1 student", "{count} students", student_count).format(count=student_word)
+        responsible_phrase = ngettext("1 responsible", "{count} responsibles", responsible_count).format(
+            count=responsible_word
         )
-        msg += create_user_list_html_string_for_message(students_created + responsibles_created)
-        self.success_messages.append(msg)
+
+        message = ngettext(
+            "Successfully created 1 course/evaluation, {student_phrase} and {responsible_phrase}",
+            "Successfully created {evaluation} courses/evaluations, {student_phrase} and {responsible_phrase}",
+            len(self.evaluations),
+        ).format(responsible_phrase=responsible_phrase, student_phrase=student_phrase, evaluation=len(self.evaluations))
+        if student_word != "no" or responsible_word != "no":
+            message += ":"
+            message += create_user_list_html_string_for_message(students_created)
+        self.success_messages.append(message)
 
     def create_test_success_messages(self):
         filtered_users = [user_data for user_data in self.users.values() if not user_data.user_already_exists()]
 
         self.success_messages.append(_("The test run showed no errors. No data was imported yet."))
         msg = format_html(
-            _("The import run will create {} courses/evaluations and {} users:"),
-            len(self.evaluations),
-            len(filtered_users),
+            ngettext(
+                "The import run will create 1 course/evaluation",
+                "The import run will create {evaluation} courses/evaluations",
+                len(self.evaluations),
+            ).format(evaluation=len(self.evaluations))
         )
-        msg += create_user_list_html_string_for_message(filtered_users)
+        if len(filtered_users) == 0:
+            msg += format_html(_(" and no users."))
```

Comment:
I think I like the `*_phrase` approach from the other method better than adding these strings together. Ignoring translation for a second, you could have something like `msg = f"The import run will create {evaluations_phrase} and {users_phrase}"` here. What do you think?